core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -94,8 +88,20 @@ class LogSegment private[log] (val log: FileRecords,
   private var rollingBasedTimestamp: Option[Long] = None
 
   /* The maximum timestamp we see so far */
-  @volatile private var maxTimestampSoFar: Long = timeIndex.lastEntry.timestamp
-  @volatile private var offsetOfMaxTimestamp: Long = timeIndex.lastEntry.offset
+  @volatile private var _maxTimestampSoFar: Long = -1L
+  def maxTimestampSoFar: Long = {
+    if (_maxTimestampSoFar < 0) {
+      _maxTimestampSoFar = timeIndex.get.lastEntry.timestamp
+    }
+    _maxTimestampSoFar
+  }
+  @volatile private var _offsetOfMaxTimestamp: Long = -1L
+  def offsetOfMaxTimestamp: Long = {
+    if (_offsetOfMaxTimestamp < 0) {
+      _offsetOfMaxTimestamp = timeIndex.get.lastEntry.offset
+    }
```

Comment:
No need for {} for single line statement.

Reply:
Done.